### PR TITLE
[Serverless] merge extension v9 into main

### PR DESCRIPTION
**Pull request by maxday**
### What does this PR do?

Merge the lambda extension release branch 9 into main
Here is the full list of released features : https://github.com/DataDog/datadog-lambda-extension/releases/tag/v9

### Checklist

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

**Review by remeh — APPROVED** [review]
LGTM! 👍 
Before merging, please make sure the licenses are updated accordingly (see the CI).

**Review by nhinsch — APPROVED** [review]

File: pkg/trace/agent/agent.go [hunk 1]
```diff
@@ -460,3 +460,8 @@ func newEventProcessor(conf *config.AgentConfig) *event.Processor {
 
 	return event.NewProcessor(extractors, conf.MaxEPS)
 }
+
+// SetGlobalTags sets global tags to the agent configuration
```
Comment (gbbr) [inline pkg/trace/agent/agent.go:464]: Is there any risk of a race here? Seem on the edge...

Comment (maxday) [reply pkg/trace/agent/agent.go:464]: It's indeed possible to receive logs before calling this method. Maybe I can add a comment to make it clearer ?

Comment (gbbr) [reply pkg/trace/agent/agent.go:464]: Sorry but I'm not sure what you mean. The question is whether there is a risk of a race condition, where the config object gets initialised and this method gets called at the same time, resulting in a panic and a crash.

Comment (maxday) [reply pkg/trace/agent/agent.go:464]: Yes I see. In the way we are using it, we are 100% sure that config has already been initialized

Comment (gbbr) [reply pkg/trace/agent/agent.go:464]: Still, this is bad API and exposes that risk. I don't feel comfortable with this solution. You could instead just set the tags directly on the config object in `cmd/serverless/main.go` around this line: https://github.com/DataDog/datadog-agent/blob/eabed0840342392f9cd118ecdb9f58ea930a8f8b/cmd/serverless/main.go#L355
 
What am I missing?

**Review by gbbr — CHANGES_REQUESTED** [review]

File: pkg/trace/agent/agent.go (hunk 1 above)
Comment (maxday) [reply pkg/trace/agent/agent.go:464]: Yes we did that before and we needed to know tags in advance, during the log agent setup, which is before the actual first lambda invocation. To retrieve those tags, we needed a STS call. We have removed that for performance issues. Instead we are waiting for the first invocation to occur, which give us the ARN. From this ARN we can get tags without having to call STS. Then we apply those tags to the agent config + we make sure that no traces go through the agent before setting those tags

Comment (maxday) [reply pkg/trace/agent/agent.go:464]: Hi @gbbr did you have the chance to look at my previous comment? Let me know what do you think about it.

**Review by gbbr — APPROVED** [review]

**Pull request merged**